fix(docker): move commands to build

herkulano opened the pull request:
move `git clone` and `yarn build` commands to `docker build`

afolarin commented:
@herkulano  Cool.. looks like it builds ok. 

I was wondering if we should have a separate build on dockerhub tracking containers that would be produced for each of the BRANCH='xxxx' 
- track different branches (seems a bit messy)
- track different sub-dirs (with the different dockerfiles)

herkulano commented:
@afolarin I've changed it to use the files from the build context, instead of using `git clone`, so we no longer need an `ENV BRANCH` variable.

I've changed docker hub to also build tags with this regexp: `/[0-9.]+/`, so all tags that have a version number will be built to docker hub. 

With this feature, we can have stable builds, e.g, `1.2.1`, and other types of builds, e.g., `1.2.3-beta`,  `1.2.3-rc`, `1.2.3-experimental`, etc.

https://github.com/RADAR-CNS/RADAR-Dashboard/pull/73
